app/controllers/comments_controller.rb

```diff
@@ -22,15 +22,17 @@ def index
         Organization.find_by(slug: params[:username]) ||
         not_found
       @commentable = @root_comment&.commentable ||
-        @user.articles.find_by(slug: params[:slug]) ||
-        not_found
+        @user.articles.find_by(slug: params[:slug]) || nil
       @article = @commentable
-      not_found unless @commentable.published
+      if @commentable
+        not_found unless @commentable.published
+      end
```

Comment:
this can be refactored to:

```ruby
not_found unless @commentable&.published
```

Reply:
Hi @rhymes, thanks for reviewing. I initially had the same thought but that wouldn't work as it would return not_found unless there is a @commentable. I can refactor to: 
```ruby
not_found if @commentable && !@commentable.published
```